components.md

```diff
@@ -0,0 +1,130 @@
+This is the place where all instrument libraries (including pymeasure) wire their hardware interface classes into ECP.
+
+Concerning the ECP, we draw the abstraction boundary at the Driver -- the details on how this communicates with a Device (SCPI, dll, ...) should not be relevant for the protocol details.
+What we define is how the other ECP components interact with the Driver, how it determines and announces its capabilities, etc.
+
+A driver must contain/manage
+* An object that communicates with the connected Device, including managing its lifecycle (init, operations, shutdown)
+* The name of the connected instrument
+* A list of available Parameters (properties/attributes to get/set) and Actions(methods to call)
+* `set`/`get`/`get_all`/`call` interfaces (for incoming commands to use to act on Parameters and Actions)
+
+A driver may contain/manage
+* A list of Parameters to poll/publish regularly (and the interval for that)
+* A cache of parameter values (to avoid unnecessary communication). 
+    - If caching is included, the `get*` interfaces must include a configurable cache timeout and a way to force fetching a fresh value. 
+* Concurrent access management/locking
+* Logging configuration
+
+:::{admonition} TODO
+We might want to add the notion of "Channels", especially for the multi-Director stuff
+:::
+
+## Parameter
+A Parameter is a property (in the English, not the Pythonic sense) of the device represented by a Driver
+It has a name and can be read(`get`) or `set`.
+
+:::{note}
+Recent values may be cached in the Driver.
+:::
+
+It may correspond closely to _attributes_ or Python (or PyMeasure) _properties_ of the instrument interface classes.
+It may have unit information, that is used when sending data over the network.
+
+## Procedures
+Sequences of steps that make up experiment runs, e.g. PyMeasure procedures.
+These instructions could be consumed by a Director and trigger a sequence of commands (ramps, loops, conditionals,...).
+
+:::{admonition} TODO
+This is a placeholder, we have not fleshed out the concept yet
+:::
+
+## Coordinator
+A component primarily concerned with routing/coordinating the message flow between other components.
+It represent the intermediate zmq brokers, proxies or somesuch.
+Multiple coordinator instances may be necessary for large deployments, but a single coordinator instance should be sufficient for operation.
+
+The presence of a coordinator should avoid the complexity/scaling of a purely point-to-point approach. 
+Currently, this corresponds to the zmq majordomo broker, and/or a zmq proxy.
+
+## Observer
+A component that receives data from other components, e.g. for logging, storage, or plotting, either directly in a streaming fashion, batched, or delayed.
+It only consumes message streams, but does not command `actors`.
+
+:::{note} Depending on setup, some commanding might be necessary, e.g. to subscribe/register.
+:::
+
+## Notes 
+### Actor Driver separation
```

Comment:
I would like to update this section once we have agreed on what actually Drivers, Actors, and maybe Processors should do. Currently, it does not really represent what I wanted to say when I wrote it, especially regarding how the low-entry-threshold would be achieved, and a potential traversal of building a system might happen, at least with the current terminology.

Reply:
Sure, maybe it will be folded into the different sections and the rest discarded.

Reply:
This became streamlined into a new section "Complexity scaling", maybe this expresses your intention better now, and we can iterate/improve on that?